BitmovinAnalyticsCollector/Classes/analytics/BitmovinAnalytics.swift

```diff
@@ -25,7 +25,7 @@ public class BitmovinAnalytics {
      */
     public func detachPlayer() {
         eventDataDispatcher.disable()
-        stateMachine.reset()
+        stateMachine = StateMachine(config: self.config)
```

Comment:
what is the thought behind this change? With it, we seem to be leaking StateMachine objects. Steps to reproduce are 

1) Using the sample app, start playback using either AVPlayer or Bitmovin Player. Set a break point in StateMachine.deinit() method
2) Press the reload button.
3) Notice that our break point in deinit is never called
4) Play the new stream
5) Pause the stream and wait for the next heartbeat event to occur. 
6) Notice that two heartbeat events are sent.

Reply:
I reverted the change for now.

Reply:
Do you need to reset the sequenceNumber to 0 when the reset occurs? Other than that, it looks good!